### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by HoffsMH**
additional task on: https://jira.blueapron.com/browse/INTRNLTLS-77

Looking to increase test coverage as much as possible before publishing to rubygems

### CODE REVIEW

**Comment by HoffsMH** [comment]
initial stab.. working on rubocop issues! plan to cover more than `table_manager_service.rb` later on in this pr.

File: spec/eventosaurus/workers/sidekiq_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+RSpec.describe Eventosaurus::Workers::Sidekiq do
+  class MockEvent
+    include Eventosaurus::Storable
+  end
```
Comment (HoffsMH) [inline spec/eventosaurus/workers/sidekiq_spec.rb]: Im worried this is an anti-pattern. Is there a better way to do this?

File: spec/eventosaurus/services/table_manager_service_spec.rb [hunk 2]
```diff
@@ -1,13 +1,86 @@
...
+      before do
+        allow(Eventosaurus::TableManagerService).to(
+          receive(:dynamodb_client) { mock_client }
+        )
+        allow(Eventosaurus).to(
+          receive(:event_table_definitions) { mock_table_to_be_created }
+        )
+      end
+
+      it 'creates the tables' do
```
Comment (HoffsMH) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: Should maybe be more descriptive like it 'tells the aws client to create the tables'?

Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: sounds good

File: spec/eventosaurus/services/table_manager_service_spec.rb [hunk 3]
```diff
@@ -1,13 +1,86 @@
...
+    [{ name: :a_single_table_to_be_created,
+       partition_key: { pkey: :s },
+       local_indexes: { skey: :n, event_uuid: :s, created_at: :s },
+       full_table_name: 'test_a_single_table_to_be_created' }]
+  end
+
+  let!(:mock_list_already_exists) do
+    instance_double('Eventosaurus.configuration.dynamodb_client.list_tables',
+       table_names: ['test_a_single_table_that_already_exists'])
+  end
```
Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: can you put these let blocks inside the context where they will be used

Comment (HoffsMH) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: @keithmgould They are used in multiple contexts. Are we ok with the code duplication?

Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: gotcha NM

File: spec/eventosaurus/services/table_manager_service_spec.rb [hunk 4]
```diff
@@ -28,4 +101,56 @@
...
+
+      it 'deletes the table' do
+        subject
+        expect(mock_client).to have_received(:delete_table).once
+      end
+    end
+  end
+
+  describe '.describe_tables' do
+    subject { Eventosaurus::TableManagerService.describe_tables }
```
Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: empty line between subject and next line (same for subject lines above)

File: spec/eventosaurus/services/table_manager_service_spec.rb [hunk 5]
```diff
@@ -28,4 +101,56 @@
...
+        allow(Eventosaurus).to(
+          receive(:event_table_definitions) { mock_table_already_exists }
+        )
+      end
+
+      it 'describes the table' do
+        table_name = mock_table_already_exists.first[:full_table_name]
+
+        expect(subject.length).to eq(1)
+        expect(subject.first[:table_name]).to eq(table_name)
```
Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: can we just do:

`
expect(subject).to eq([table_name])
`

?

File: spec/eventosaurus/workers/sidekiq_spec.rb (hunk 1 above)
Comment (keithmgould) [inline spec/eventosaurus/workers/sidekiq_spec.rb]: what seems wrong about this?

File: spec/eventosaurus/services/table_manager_service_spec.rb (hunk 5 above)
Comment (HoffsMH) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: @keithmgould the hash returned from `subject` has more keys in it than just `:table_name`. I thought it was sensible to at least, make sure that the names matched. will calling `eq()` work regardless?

File: spec/eventosaurus/services/table_manager_service_spec.rb (hunk 4 above)
Comment (HoffsMH) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: addressed (I think?)

File: spec/eventosaurus/services/table_manager_service_spec.rb (hunk 5 above)
Comment (keithmgould) [inline spec/eventosaurus/services/table_manager_service_spec.rb]: gotcha. NM

**Comment by keithmgould** [comment]
take a look at difference between `let!()` and `let()`.  Did you mean to do `let!` ?

**Comment by HoffsMH** [comment]
@keithmgould a lot of the values need to be loaded  before we ever enter the it section so I thought it best for them to be eager but my understanding of this works is still limited. I can go back and get rid of every bang that isn't needed.

File: spec/eventosaurus/workers/sidekiq_spec.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      end
+    end
+
+    context 'when AWS does raise an error' do
+      before do
+        MockEvent = class_spy('SpecEvent')
+
+        allow(MockEvent).to(
+          receive(:put_item)
+            .and_raise(Aws::DynamoDB::Errors::ConditionalCheckFailedException.new(1, 2))
```
Comment (keithmgould) [inline spec/eventosaurus/workers/sidekiq_spec.rb]: wonder if this warrants a comment, it looks pretty strange.

File: spec/eventosaurus/persistors/synchronous_spec.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+        expect(SpecEvent).to have_received(:put_item)
+      end
+    end
+
+    context 'when AWS does raise an error' do
+      before do
+        allow(SpecEvent).to(
+          receive(:put_item)
+            .and_raise(Aws::DynamoDB::Errors::ConditionalCheckFailedException.new(1, 2))
```
Comment (keithmgould) [inline spec/eventosaurus/persistors/synchronous_spec.rb]: please add a note above this line. Maybe: "due to how `Aws::DynamoDB::Errors::ConditionalCheckFailedException` is coded, we must instantiate the class with (random) arguments as follows"

**Comment by keithmgould** [comment]
just left a comment about leaving a comment. once thats done consider this a 👍 and feel free to merge!

**Pull request merged**